server/core/src/https/mod.rs

```diff
@@ -429,568 +110,475 @@ pub async fn create_https_server(
-        the CompressMiddleware configuration.
-
-        The attack targets secrets compressed and encrypted in flight with the intent
-        to infer their content.
-
-        This is not a concern for the paths covered by this configuration
-        ( /, /ui/<and all sub-paths>, /pkg/<and all sub-paths> ),
-        as they're all static content with no secrets in transit - all that data should
-        come from Kanidm's REST API, which is on a different path and not covered by
-        the compression middleware.
-
-
-        [1] - https://resources.infosecinstitute.com/topic/the-breach-attack/
-        */
-
-        let compress_middleware = CompressMiddleware::builder()
-            .threshold(1024)
-            .content_type_check(Some(compression_content_type_checker()))
-            .build();
-
-        let mut static_tserver = tserver.at("");
-        static_tserver.with(StaticContentMiddleware::default());
-
-        static_tserver.with(UIContentSecurityPolicyResponseMiddleware::new(js_files));
-
-        // The compression middleware needs to be the last one added before routes
-        static_tserver.with(compress_middleware.clone());
-
-        static_tserver.at("/").mapped_get(&mut routemap, index_view);
-        static_tserver
-            .at("/robots.txt")
-            .mapped_get(&mut routemap, robots_txt);
-        static_tserver
-            .at("/manifest.webmanifest")
-            .mapped_get(&mut routemap, manifest);
-        static_tserver
-            .at("/ui/")
-            .mapped_get(&mut routemap, index_view);
-        static_tserver
-            .at("/ui/*")
-            .mapped_get(&mut routemap, index_view);
+        js_files: get_js_files(role.clone()),
+    };
 
-        let mut static_dir_tserver = tserver.at("");
-        static_dir_tserver.with(StaticContentMiddleware::default());
-        // The compression middleware needs to be the last one added before routes
-        static_dir_tserver.with(compress_middleware);
-        static_dir_tserver
-            .at("/pkg")
-            .serve_dir(env!("KANIDM_WEB_UI_PKG_PATH"))
-            .map_err(|e| {
-                error!(
-                    "Failed to serve pkg dir {} -> {:?}",
-                    env!("KANIDM_WEB_UI_PKG_PATH"),
-                    e
+    let app = match role {
+        ServerRole::WriteReplica | ServerRole::ReadOnlyReplica => {
+            let pkg_path = PathBuf::from(env!("KANIDM_WEB_UI_PKG_PATH"));
+            if !pkg_path.exists() {
+                eprintln!(
+                    "Couldn't find Web UI package path: ({}), quitting.",
+                    env!("KANIDM_WEB_UI_PKG_PATH")
                 );
-            })?;
+                std::process::exit(1);
+            }
+            Router::new()
+                .route("/", get(|| async { Redirect::temporary("/ui/login") }))
+                .route("/ui", get(crate::https::ui::ui_handler))
+                .route("/ui", get(crate::https::ui::ui_handler))
```

Comment:
should this second ui route have a trailing slash like this: `"/ui/"` ?

Reply:
Good catch - turns out I needed to add *another* route to cover that, because `/ui` and `/ui/*key` weren't enough.